tmt/base.py

```diff
@@ -2723,6 +2724,7 @@ def __init__(self,
         self.inject_logger(logger)
 
         self._path = path or Path.cwd()
+        self._name = name or ''
```

Comment:
Can the "unset" meaning be expressed with `None`? The annotation says `Optional[str]`, and `None` is easier to track than an empty string.

Reply:
Yes, I tried to do it originally, but `Optional[str]` is in conflict with [`Common.name`](https://github.com/teemtee/tmt/blob/main/tmt/utils.py#L1553), which is `str`. Any ideas for this? I could just rename this attribute to something else other than `self._name`, but IMO it is a reasonable name.

Reply:
I would consider renaming, or rather adding a new attribute. It's clear that, sometimes, a tree will have no value for this attribute, e.g. `Tree.init()` creates one, `tmt.cli.main()` creates one that would be hard to name, etc. Do I understand it correctly you want to use it to propagate the phase name to tests, to be shown/used somewhere as needed - that sounds like a dedicated attribute might fit the goal better, to convey the "name of the phase that spawned me" value.

Also, something to consider, it may happen that one fmf tree is used by multiple `discover` phases that would, e.g. with different filters. It would be still the very same fmf tree, `tmt.Tree` instances created from it would be virtually the same, but with `name`, we would give it different names, which does not seem right - on the other hand, it may very well have a different "name of the phase that spawned me" label.